jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2368,21 +2377,58 @@ private ClassDefiner classDefiner(byte[] bytes, ClassOption... classOptions) thr
 				throw new IllegalAccessException();
 			}
 
+			int flags = 0;
+			for (ClassOption opt : classOptions) {
+				flags |= opt.toFlag();
+			}
+			MethodHandleNatives.checkClassBytes(bytes);
```

Comment:
My proposal was that this method return the classname as a String to avoid partially reparsing the classfile.  The checks in `getClassName` can still be done in java code after renaming the method to something like `validateClassName`.  
```suggestion
			String targetClassName = MethodHandleNatives.checkClassBytes(bytes);
			validateClassName(targetClassName, bytes);
```